geomesa-core/src/main/scala/org/locationtech/geomesa/core/data/AccumuloFeatureReader.scala

```diff
@@ -20,32 +20,24 @@ import org.geotools.data.{FeatureReader, Query}
 import org.locationtech.geomesa.core.index._
 import org.locationtech.geomesa.core.stats._
 import org.locationtech.geomesa.core.util.ExplainingConnectorCreator
+import org.locationtech.geomesa.feature.FeatureEncoding.FeatureEncoding
 import org.locationtech.geomesa.feature.SimpleFeatureEncoder
 import org.locationtech.geomesa.utils.stats.{MethodProfiling, TimingsImpl}
 import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
 
 class AccumuloFeatureReader(dataStore: AccumuloDataStore,
                             query: Query,
-                            indexSchemaFmt: String,
                             sft: SimpleFeatureType,
-                            featureEncoder: SimpleFeatureEncoder)
+                            indexSchemaFmt: String,
+                            featureEncoding: FeatureEncoding,
+                            version: Int)
     extends FeatureReader[SimpleFeatureType, SimpleFeature] with MethodProfiling {
 
   implicit val timings = new TimingsImpl
 
-  private val indexSchema = IndexSchema(indexSchemaFmt, sft, featureEncoder)
-  private val queryPlanner = indexSchema.planner
-  private lazy val strategyHints = dataStore.strategyHints(sft)
-
-  def explainQuery(o: ExplainerOutputType = ExplainPrintln) = {
-    profile({
-      val cc = new ExplainingConnectorCreator(o)
-      queryPlanner.getIterator(cc, sft, query, strategyHints, o)
-    }, "explain")
-    o(s"Query Planning took ${timings.time("explain")} milliseconds.")
-  }
-
-  private lazy val iter = profile(queryPlanner.query(query, dataStore, strategyHints), "planning")
+  private val hints = dataStore.strategyHints(sft)
+  private val planner = new QueryPlanner(sft, featureEncoding, indexSchemaFmt, dataStore, hints, version)
+  private val iter = profile(planner.query(query), "planning")
```

Comment:
Definitely for another sprint . . . can you add a ticket about hooking up dropwizard metrics into this profile stuff here?